### PR TITLE
[Feature] Extend supported file types #1

### PR DESCRIPTION
**Pull request by alexstocker**
- #1 Added basic OBJ support
- Added light and ambient
- Added default material
- #1 Added basic FBX support
- Added basic debugging mode
- #1 Added basic STL support

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -33,6 +33,18 @@ export default defineWebApplication({
...
+            extension: 'stl',
+            label: 'View STL 3D Model'
+          },
+          {
+            extension: 'fbx',
+            label: 'View FBX 3D Model'
+          },
+          {
+            extension: 'obj',
+            label: 'View OBJ 3D Model'
```
Comment (saw-jan) [inline src/index.ts:47]: general label should be fine.
```suggestion
            label: 'View 3D Model'
          },
          {
            extension: 'fbx',
            label: 'View 3D Model'
          },
          {
            extension: 'obj',
            label: 'View 3D Model'
```

File: src/App.vue [hunk 2]
```diff
@@ -286,38 +365,54 @@ function updateLocalHistory() {
...
 
   updateLocalHistory()
+  await unloadModels()
   // TODO: how to prevent activeFiles from being reduced
   // load activeFiles
   await loadFolderForFileContext(unref(currentFileContext))
   await renderNewModel()
 }
+
+async function unloadModels(): Promise<void> {
```
Comment (saw-jan) [inline src/App.vue:407]: could you also check how we can include this as well? or may not be required with this new method
https://github.com/saw-jan/web-app-3dmodel-viewer/blob/bb668d1586ebbb904af5a744650d939a1e06c4a0/src/App.vue#L241

you are right, the cleanup stuff should have it's own method

File: src/App.vue [hunk 3]
```diff
@@ -182,6 +192,9 @@ const modelFiles = computed<Resource[]>(() => {
   return sortHelper(files, [{ name: unref(sortBy) }], unref(sortBy), unref(sortDir))
 })
 const activeModelFile = computed(() => unref(modelFiles)[unref(activeIndex)])
+const pageTitle = computed(() => `Preview for ${unref(activeModelFile)?.name}`)
+const fileId = computed(() => unref(currentFileContext).itemId)
```
Comment (saw-jan) [inline src/App.vue:196]: this should be left over from conflict resolution. these are not required anymore
```suggestion
```

File: src/App.vue [hunk 4]
```diff
@@ -340,6 +436,15 @@ function resetModelPosition() {
     camera.lookAt(iniCamPosition)
   }
 }
+
+function debug(output) {
+  if (debugIsEnabled) {
+    scene.add(new AxesHelper(10))
+    console.log('####### DEBUG #######')
```
Comment (saw-jan) [inline src/App.vue:443]: would be good to add context in the log as well.
```suggestion
    console.log('####### DEBUG 3D MODEL #######')
```

File: src/App.vue (hunk 2 above)
Comment (alexstocker) [reply src/App.vue:404]: Removed scene.remove from renderNewModel. no longer needed.

**Review by saw-jan — APPROVED** [review]
Looks good.
I tested and some files were not opened or atleast properly but for now it is good to be merged.
Thanks for your effort and time. :heart:

**Comment by alexstocker** [comment]
@saw-jan could you please provide files which fail to open? I would like to investigate on it.

**Comment by saw-jan** [comment]
> @saw-jan could you please provide files which fail to open? I would like to investigate on it.

it was just too big that's why I was not able to see it at first. But I will send you the files

**Comment by saw-jan** [comment]
Here is the model that was too big and only visible when rotating. Also, there's some problem with auth
[Mclaren.zip](https://github.com/user-attachments/files/17361212/Mclaren.zip)

**Comment by alexstocker** [comment]
> Here is the model that was too big and only visible when rotating. Also, there's some problem with auth [Mclaren.zip](https://github.com/user-attachments/files/17361212/Mclaren.zip)

@saw-jan Thanx!

**Model size**
we could think about re-scaling the model on the fly if suitable and/or get rid of hard coded min and max orbit distances 

**File size**
when it comes to support different types of 3D standards we are going to face rendering limitations by the browser or better to say the underlying hardware resources and the support of theme (os depending), pretty soon (keywords: webgl, level of detail, shading, lights ...). Even if a huge file could be loaded into memory, this does not mean that the performance of rendering the scene will meet the user's expectations. 

**Auth Issue**
[CONFIRM] It's pretty strange facing auth showing up on loading the model you provided. Did not notice something like this on any other model. This is maybe FBX related, but I am not sure. Still investigating.

image

**Comment by alexstocker** [comment]
@saw-jan To come to an end. It looks like missing materials/textures. Even Blender is blaming about ;-)

```
...
FBX import: Materials & Textures...
Error, could not find any file path in  FBXElem(id=b'Texture', props=[1961718568848, b'Map #2138625263\x00\x01Texture', b''], props_type=bytearray(b'LSS'), elems=[FBXElem(id=b'Type', props=[b'TextureVideoClip'],
...
```

**Result** Trying to load non existing textures from non existing texture folder causing auth request

**Comment by saw-jan** [comment]
>when it comes to support different types of 3D standards we are going to face rendering limitations by the browser or better to say the underlying hardware resources and the support of theme (os depending), pretty soon (keywords: webgl, level of detail, shading, lights ...). Even if a huge file could be loaded into memory, this does not mean that the performance of rendering the scene will meet the user's expectations.

yeah, there're a lot of tradeoffs to consider. we can only try to minimize them.

**Pull request merged**